runtime/near-vm-logic/src/errors.rs

```diff
@@ -1,29 +1,52 @@
 use crate::dependencies::ExternalError;
 
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum HostError {
     BadUTF16,
     BadUTF8,
     GasExceeded,
     GasLimitExceeded,
     BalanceExceeded,
     EmptyMethodName,
-    External(ExternalError),
     GuestPanic,
     IntegerOverflow,
-    InvalidIteratorIndex,
     InvalidPromiseIndex,
     CannotAppendActionToJointPromise,
     CannotReturnJointPromise,
     InvalidPromiseResultIndex,
     InvalidRegisterId,
     IteratorWasInvalidated,
     MemoryAccessViolation,
+    InvalidReceiptIndex,
+    InvalidIteratorIndex,
+    InvalidAccountId,
+    InvalidMethodName,
+    InvalidPublicKey,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum HostErrorOrStorageError {
```

Comment:
Why do we need this struct if it has the same content as `ExternalError`?

Reply:
ExternalError is a small set of errors, we can change External interface to this set of errors